services/net/transcription/TranscriptionManager.cs

```diff
@@ -0,0 +1,252 @@
+                catch (Exception ex)
+                {
+                    this.Logger.LogError(ex, "Service had an unexpected failure.");
+                    this.State.RecordFailure();
+                }
+            }
+
+            // The delay ensures we don't have a run away thread.
+            this.Logger.LogDebug("Service sleeping for {delay} ms", delay);
+            await Task.Delay(delay);
+        }
+    }
+
+    /// <summary>
+    /// Import the content.
+    /// Copy any file associated with source content.
+    /// </summary>
+    /// <param name="result"></param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    private async Task HandleMessageAsync(ConsumeResult<string, SourceContent> result)
+    {
+        if (this.State.Status != ServiceStatus.Running)
+        {
+            this.Consumer.Stop();
+            this.State.Stop();
+        }
+
+        this.Logger.LogInformation("Importing Content from Topic: {Topic}, Uid: {Key}", result.Topic, result.Message.Key);
+
+        // TODO: Failures after receiving the message from Kafka will result in missing content.  Need to handle this scenario.
+        // TODO: Handle e-tag.
+        var source = await _api.GetDataSourceAsync(result.Value.Source) ?? throw new InvalidOperationException($"Failed to fetch data source for '{result.Topic}'");
+
+        if (source.ContentTypeId == null) throw new InvalidOperationException($"Data source not configured to import content correctly");
+
+        var content = new ContentModel()
+        {
+            Status = Entities.ContentStatus.Draft, // TODO: Automatically publish based on Data Source config settings.
+            WorkflowStatus = Entities.WorkflowStatus.Received, // TODO: Automatically extract based on lifecycle of content reference.
+            ContentTypeId = source.ContentTypeId.Value,
+            MediaTypeId = source.MediaTypeId,
+            LicenseId = source.LicenseId,
+            SeriesId = null, // TODO: Provide default series from Data Source config settings.
+            OtherSeries = null, // TODO: Provide default series from Data Source config settings.
+            OwnerId = source.OwnerId,
+            DataSourceId = source.Id,
+            Source = result.Topic,
+            Headline = result.Message.Value.Title,
+            Uid = result.Message.Value.Uid,
+            Page = "", // TODO: Provide default page from Data Source config settings.
+            Summary = result.Message.Value.Summary,
+            Transcription = "",
+            SourceUrl = result.Message.Value.Link,
+            PublishedOn = result.Message.Value.PublishedOn,
+        };
+
+        // Only add if doesn't already exist.
+        var exists = await _api.FindContentByUidAsync(content.Uid, content.Source);
+        if (exists == null)
+        {
+            // Upload the file to the API.
+            if (content != null && !String.IsNullOrWhiteSpace(result.Message.Value.FilePath))
```

Comment:
I believe the file you're actually interested in is the one the API is using.  This will also ensure on-demand requests for transcription uses this exact same process.

Reply:
I was assuming that on-demand requests would put the location of the curated clip in result.Message.Value.FilePath in the same way the clip service does. Not sure what you mean by "the one the API is using".

Reply:
Fixed...I think.